### ISSUE TITLE
[Bug]: Text controls (and likely others) aren't connected to their labels, which prevents screenreaders from announcing them

### ISSUE THREAD

**Issue by msftedad** [storybookjs/storybook#24127]
### Describe the bug

Any HTML input should be connected to a label (or in last resort, have an `aria-label` attribute). Currently, text controls (and probably other controls) are not connected to their labels. This means that screenreaders announce the control value without its label. This makes the controls panel hard-to-impossible to use for blind users.

#### Current Result:
While navigating to “Brand” edit field narrator is not announcing the label name of the edit field.

#### Expected Result:
While navigating to “Brand” edit field narrator should announce the label name for “Brand” edit field.


### To Reproduce

1. Open a Storybook instance with stories that have text controls
2. Navigate to a story with text controls
3. Open the “Controls” panel and navigate to the value field for a text control
4. Observe the screenreader announcement

See live video example:
[label name for edit field.webm](https://github.com/storybookjs/storybook/assets/93735775/0b585ca2-a604-4bf9-a636-b8bf4734a8dd)
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               

### System

```bash
OS: Windows 11
OS version: 22H2 (OS Build 22621.2134)
Edge Version: 116.0.1938.69 (Official build) (64-bit)
```


### Additional context

_No response_

**Comment by MarioKavula** [comment]
Hello, I am new to open source and I would like to help with this issue or other accessibility-related issues.

**Comment by abhisheksharma010** [comment]
Hey there! 👋

I'm new to open source and super excited to dive in! I'd love to work on this. Can I be assigned to this issue? 🚀

**Comment by Sidnioulz** [comment]
Anyone can feel free to take this issue and work on it without assignment!

The PR opened by @MarioKavula looked correct in spirit but had not been updated for too long.  Note that I've updated the issue description a bit to align it with the vocabulary we usually use and to make it a bit clearer.
